app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt

```diff
@@ -59,69 +61,129 @@ class DashboardFragment : BaseFragment<FragmentDashboardBinding, DashboardVM>(
         rxPermissions = RxPermissions(this)
         subsribeToViewModel()
 
-        viewModel.exposureNotificationsEnabled.observe(this, Observer {
+        viewModel.exposureNotificationsEnabled.observe(this, Observer { enEnabled ->
             refreshDotIndicator()
-            if (it) {
+            if (enEnabled) {
                 LocalNotificationsHelper.dismissNotRunningNotification(context)
             }
         })
+
+        viewModel.permissionsState.observe(this, Observer { permissionsState ->
+            when (permissionsState) {
+                DashboardVM.PermissionsState.BOTH_ENABLED -> onBtAndLocationEnabled()
+                DashboardVM.PermissionsState.LOCATION_DISABLED -> onOnlyLocationDisabled()
+                DashboardVM.PermissionsState.BT_DISABLED -> onOnlyBtDisabled()
+                DashboardVM.PermissionsState.BOTH_DISABLED -> onBothBtAndLocationDisabled()
+            }
+        })
     }
 
     override fun onStart() {
         super.onStart()
-        context?.registerReceiver(btAndLocationReceiver, IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED))
+        context?.registerReceiver(
+            btAndLocationReceiver,
+            IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED)
+        )
         context?.registerReceiver(
             btAndLocationReceiver,
             IntentFilter(LocationManager.PROVIDERS_CHANGED_ACTION)
         )
+
     }
 
     override fun onStop() {
         context?.unregisterReceiver(btAndLocationReceiver)
         super.onStop()
     }
 
-    private fun refreshDotIndicator(){
-        mainViewModel.serviceRunning.value = viewModel.exposureNotificationsEnabled.value && requireContext().isBtEnabled() && requireContext().isLocationEnabled()
+    private fun refreshDotIndicator() {
+        mainViewModel.serviceRunning.value =
+            viewModel.exposureNotificationsEnabled.value && requireContext().isBtEnabled() && requireContext().isLocationEnabled()
     }
 
     private fun subsribeToViewModel() {
         subscribe(DashboardCommandEvent::class) { commandEvent ->
             when (commandEvent.command) {
                 DashboardCommandEvent.Command.DATA_UP_TO_DATE -> {
-                    LocalNotificationsHelper.dismissOudatedDataNotification(context)
-                    data_notification_container.hide()
+//                    LocalNotificationsHelper.dismissOudatedDataNotification(context)
```

Comment:
Why commented code?

Reply:
it's a WIP, I jut cannot push directly to the branch